From 7901d9086cace08659ea7e8bde52e4c464542958 Mon Sep 17 00:00:00 2001
From: Julian Ospald <hasufell@posteo.de>
Date: Sat, 1 Oct 2016 14:53:46 +0200
Subject: [PATCH] Fix picking up toolchain from environment
Upstream: submitted, https://github.com/matricks/bam/pull/103

* respect AR as well
* set settings.ddl.exe to CXX
---
 src/tools.lua | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/tools.lua b/src/tools.lua
index ce7cfeb..c62a80e 100644
--- a/src/tools.lua
+++ b/src/tools.lua
@@ -36,6 +36,11 @@ function SetDefaultDrivers(settings)
 	if os.getenv("CXX") then
 		settings.cc.exe_cxx = os.getenv("CXX")
 		settings.link.exe = os.getenv("CXX")
+		settings.dll.exe = os.getenv("CXX")
+	end
+
+	if os.getenv("AR") then
+		settings.lib.exe = os.getenv("AR")
 	end
 
 	if os.getenv("CFLAGS") then
-- 
2.10.0

